test/ruby_lsp_rails/server_test.rb

```diff
@@ -268,6 +270,79 @@ def print_it!
     $> = original_stdout
   end
 
+  test "forked processes are named based on caller" do
+    skip("Fork is not supported on Windows") if Gem.win_platform?
+
+    addon_path = File.expand_path("my_addon.rb")
+    File.write(addon_path, <<~RUBY)
+      class MyServerAddon < RubyLsp::Rails::ServerAddon
+        def name
+          "MyAddon"
+        end
+
+        def execute(request, params)
+          file = "process_name.txt"
+          pid = fork do
+            # We can't directly send a message in theses tests because we're using a StringIO as stdout instead of the
+            # actual pipe, which means that the child process doesn't have access to the same object
+            process_title = `ps -p \#{Process.pid} -o comm=`.lines.last.strip
+            File.write(file, process_title)
+          end
+
+          Process.wait(pid)
+          send_message({ process_name: File.read(file) })
+          File.delete(file)
+        end
+      end
+    RUBY
+
+    begin
+      @server.execute("server_addon/register", server_addon_path: addon_path)
+      @server.execute("server_addon/delegate", server_addon_name: "MyAddon", request_name: "dsl")
+      assert_equal(response, { process_name: "ruby-lsp-rails: #{addon_path}" })
+    ensure
+      FileUtils.rm(addon_path)
+    end
+  end
+
+  test "forked processes with no block are named based on caller" do
+    skip("Fork is not supported on Windows") if Gem.win_platform?
+
+    addon_path = File.expand_path("my_other_addon.rb")
+    File.write(addon_path, <<~RUBY)
+      class MyOtherServerAddon < RubyLsp::Rails::ServerAddon
+        def name
+          "MyOtherAddon"
+        end
+
+        def execute(request, params)
+          file = "other_process_name.txt"
+          pid = fork
+
+          if pid
+            Process.wait(pid)
+            send_message({ process_name: File.read(file) })
+            File.delete(file)
```

Comment:
Should we also assert that this process' title (the parent process) was not changed after fork?

Reply:
Yes, good idea. I added code to assert that it didn't change.